tests/x11/mac80211_hwsim.pm

```diff
@@ -0,0 +1,219 @@
+
+    type_string "# generate certificate for CA\n";
+    assert_script_run 'openssl req -x509 -new -nodes -key CA/CA.key -sha256 -days 3650 -out CA/CA.crt -subj "/"';
+
+    type_string "# generate certificate signing request for server\n";
+    assert_script_run 'openssl req -new -key server/server.key -out server/server.csr -subj "/"';
+    save_screenshot;
+
+    type_string "# sign csr with the key/cert from the CA\n";
+    assert_script_run 'openssl x509 -req -in server/server.csr -CA CA/CA.crt -CAkey CA/CA.key -CAcreateserial -out server/server.crt -days 3650 -sha256';
+    save_screenshot;
+}
+
+sub configure_hostapd {
+    type_string "# configure hostapd\n";
+    assert_script_run 'wget -O /etc/hostapd.conf ' . data_url('hostapd_wpa2-enterprise.conf');
+
+    type_string "# create wpa2 enterprise user\n";
+    assert_script_run 'echo \"franz.nord@example.com\" PEAP >> /etc/hostapd.eap_user';
+    assert_script_run 'echo \"franz.nord@example.com\" MSCHAPV2 \"nots3cr3t\" [2]>> /etc/hostapd.eap_user';
+}
+
+sub configure_system {
+    my $self = shift;
+    y2x11test::launch_yast2_module_x11;
+    $self->yast_search('Network Settings');
+
+    # assert_and_click 'yast2-networksettings'
+    assert_and_click 'yast2_control-center_network-check';
+    # assert 'networksettings_open'
+    assert_screen 'yast2_control-center_network-opened';
+    # aac 'global options'
+    assert_and_click 'yast2_network-global_options-click';
+    # aac 'network setup method'
+    assert_and_click 'yast2_network-nm_selection-click';
+    # aac 'NetworkManager
+    assert_and_click 'yast2_network-network_manager-click';
+    assert_screen 'yast2_network-network_manager-selected';
+    assert_and_click 'yast2_network-apply_settings-click';
+    # aac 'ok'
+    assert_and_click 'yast2_network-applet_warning-click';
+    assert_screen 'yast2_network-is_applying';
+    # assert 'saving-is-running'
+    # softfail yast2 error
+    #   aac 'ok'
+    if (check_screen 'yast2_network-error_dialog') {
+        record_soft_failure 'bsc#TODO';
+        assert_and_click 'yast2_network-error_dialog';
+    }
+}
+
+sub reload_services {
+    type_string "# reload required services\n";
+    assert_script_run 'systemctl restart NetworkManager';
+    assert_script_run 'systemctl restart hostapd';
+}
+
+sub yast_search {
+    my ($self, $name) = @_;
+    # on openSUSE we have a Qt setup with keyboard shortcut
+    if (check_var('DISTRI', 'opensuse')) {
+        send_key 'alt-s';
+    }
+    # with the gtk interface we have to click as there is no shortcut
+    elsif (check_var('DISTRI', 'sle')) {
+        assert_screen([qw(yast2_control-center_search_clear yast2_control-center_search)], no_wait => 1);
+        if (match_has_tag 'yast2_control-center_search') {
+            assert_and_click 'yast2_control-center_search';
+        }
+        else {
+            assert_and_click 'yast2_control-center_search_clear';
+        }
+    }
+    type_string $name if $name;
+}
```

Comment:
In case you just copied this from the yast2_control_center test better extract it into a common base class